kubernetes/utils/deserialize.py

```diff
@@ -0,0 +1,285 @@
+
+        return obj
+
+
+def load_from_yaml(yaml_file, klass=None, verbose=False):
+    """
+    Perform an action from a yaml file. Pass True for verbose to
+    print additional information.
+    Input:
+    yaml_file: string. Contains the path to yaml file.
+    klass: class literal for deserialized object,
+    or string of class name.
+    verbose: If True, print information about the object lookup info.
+        Default is False.
+
+    Raises:
+        FailToLoadError which holds list of load failures.
+    """
+    with open(path.abspath(yaml_file)) as f:
+        yml_document_all = yaml.safe_load_all(f)
+        failures = []
+        objs = []
+        for yml_document in yml_document_all:
+            try:
+                obj = load_from_dict(
+                    yml_document, klass=klass, verbose=verbose)
+                # Prevent returning list of lists when doing multi
+                # document yaml
+                if isinstance(obj, list):
+                    objs.extend(obj)
+                else:
+                    objs.append(obj)
+
+            except FailToLoadError as failure:
+                failures.extend(failure.load_exceptions)
+
+        if objs:
+            result = objs if len(objs) > 1 else objs[0]
+
+        if failures:
+            raise FailToLoadError(failures)
+
+        return result
+
+
+def response_type_from_dict(data, verbose=False):
+    """
+    Helper function to do source code lookup and extract the
+    response_type of an kubernetes object.
+    Input:
+    data: a dictionary holding a valid kubernetes object
+    verbose: If True, print confirmation from the create action.
+        Default is False.
+
+    Raises:
+        FailToLoadError which holds list of load failures.
+    """
+    response_type = None
+    failures = []
+
+    group, _, version = data["apiVersion"].partition("/")
+    if version == "":
+        version = group
+        group = "core"
+    # Take care for the case e.g. api_type is "apiextensions.k8s.io"
+    # Only replace the last instance
+    group = "".join(group.rsplit(".k8s.io", 1))
+    # convert group name from DNS subdomain format to
+    # python class name convention
+    group = "".join(word.capitalize() for word in group.split("."))
+    fcn_to_call = "{0}{1}Api".format(group, version.capitalize())
+    k8s_api = getattr(client, fcn_to_call)
+    # Replace CamelCased action_type into snake_case
+    kind = data["kind"]
+    kind = re.sub("(.)([A-Z][a-z]+)", r"\1_\2", kind)
+    kind = re.sub("([a-z0-9])([A-Z])", r"\1_\2", kind).lower()
+    # Expect the user to load namespaced objects more often
+    if hasattr(k8s_api, "read_namespaced_{0}_with_http_info".format(kind)):
+        fnc_lookup = "read_namespaced_{0}_with_http_info".format(kind)
+    elif hasattr(k8s_api, "read_{0}_with_http_info".format(kind)):
+        fnc_lookup = "read_{0}_with_http_info".format(kind)
+    # Try with the create if no read fnc found
```

Comment:
is there any API that supports CREATE but doesn't support GET?

Reply:
I think this is for objects that don't have read_xxx  functions; I believe the get_{0}_with_http_info would be more appropriate to avoid confusion.

Reply:
I can't think of an API that has `create_{0}_with_http_info` but doesn't have `read_{0}_with_http_info`/`read_namespaced_{0}_with_http_info`. Could you give an example?